prov/util/src/util_cq.c

```diff
@@ -252,57 +233,43 @@ ssize_t ofi_cq_readfrom(struct fid_cq *cq_fid, void *buf, size_t count,
 		}
 	}
 
-	if (count > ofi_cirque_usedcnt(cq->cirq))
-		count = ofi_cirque_usedcnt(cq->cirq);
+	/* FIXME: current code does not handle count > 1 */
+	if (count)
+		count = 1;
```

Comment:
The util_cq_read_xxx calls *do* increment the buf parameter, which was related to the reason I flagged this.  (I wasn't seeing buf being updated).  So I think this change can be reverted.